bcipy/display/rsvp/display.py

```diff
@@ -5,175 +5,242 @@
 from psychopy import core, visual
 
 from bcipy.acquisition.marker_writer import NullMarkerWriter, MarkerWriter
-from bcipy.helpers.task import SPACE_CHAR
+from bcipy.helpers.task import SPACE_CHAR, get_key_press
+from bcipy.display import Display
 from bcipy.helpers.stimuli import resize_image
 from bcipy.helpers.system_utils import get_screen_resolution
 from bcipy.helpers.triggers import TriggerCallback, _calibration_trigger
 
 
-class RSVPDisplay(object):
+class StimuliProperties:
+
+    def __init__(
+            self,
+            stim_font: str,
+            stim_pos: Tuple[float],
+            stim_height: float,
+            stim_inquiry: List[str],
+            stim_colors: List[str],
+            stim_timing: List[float],
+            is_txt_stim: bool):
+        self.stim_font = stim_font
+        self.stim_pos = stim_pos
+        self.stim_height = stim_height
+        self.stim_inquiry = stim_inquiry
+        self.stim_colors = stim_colors
+        self.stim_timing = stim_timing
+        self.is_txt_stim = is_txt_stim
+        self.stim_length = len(self.stim_inquiry)
+        self.sti = None
+
+    def build_init_stimuli(self, window):
+        if self.is_txt_stim:
```

Comment:
Alternately we could model this with separate subclasses (TextStimuliProperties and ImageStimuliProperties), using the parent to dispatch to the correct one. However, I'm not sure if there are any advantages to that approach at this point.

Reply:
I'll make another ticket for this! I like the idea of asking the task to set up these properties and making the display less clever.